upstairs/src/client.rs

```diff
@@ -1082,7 +1082,14 @@ impl DownstairsClient {
                     | DsState::Replay
                     | DsState::LiveRepair
                     | DsState::LiveRepairReady
+                    | DsState::Offline
```

Comment:
Is this transition actually a thing?  It looks like we added an `Offline` → `Faulted` transition, not directly `Offline` → `Deactivated`

Reply:
Yeah, if we are offline and there are no jobs in the work queue, we will go from `Offline` to `Deactivated`

Reply:
Got it, I see that transition happens before the new `Offline` → `Faulted` transition